### PR TITLE
Add update-notifier for dev mode

### LINKED ISSUE #293: Now-style update warnings
We should warn that your Next version is not up to date.

- For `next` (ie: dev mode only).
- We should only do this if a new semver patch or feature is available, but not major?
- Now-style: https://github.com/zeit/now/blob/master/lib/check-update.js (very slim)
- Not sure what this looks like with #264?

### PR DESCRIPTION
**Pull request by timneutkens**
Fixes #293

### CODE REVIEW

File: bin/next-dev [hunk 1]
```diff
@@ -5,9 +5,15 @@ import parseArgs from 'minimist'
 import { existsSync } from 'fs'
 import Server from '../server'
 import { printAndExit } from '../lib/utils'
+import pkg from '../../package.json'
+import updateNotifier from 'update-notifier'
 
 process.env.NODE_ENV = process.env.NODE_ENV || 'development'
 
+if (process.env.NODE_ENV === 'development') {
```
Comment (IanMitchell) [inline bin/next-dev]: Is this check needed? (Doesn't `next-dev` start the server in development mode?)

**Comment by arunoda** [comment]
@timneutkens did you check how this looks like.
Our use of Webpack Friendly error plugin might clear this notice.

**Comment by arunoda** [comment]
So, how about notifying via the OS level notification system?

**Comment by arunoda** [comment]
I noticed `update-notifier`s notifications are not timely sometimes.
I tend to do directly use [`configstore`](https://www.npmjs.com/package/configstore) and check for a new version directly.

File: bin/next-dev (hunk 1 above)
Comment (arunoda) [inline bin/next-dev]: I too think, we don't need this check.

Comment (timneutkens) [inline bin/next-dev]: Removed 👍

**Comment by timneutkens** [comment]
> did you check how this looks like. Our use of Webpack Friendly error plugin might clear this notice.
I have thought about it. Since the output is cleared by webpack friendly errors. I couldn't test it since we obviously don't have a new version. Though `update-notifier` adds the message on process exit. So that should be fine 👍 

>So, how about notifying via the OS level notification system?
I guess we can do both, though I prefer it as an unobtrusive message.

>I noticed update-notifiers notifications are not timely sometimes. I tend to do directly use configstore and check for a new version directly.
Could it be because of https://www.npmjs.com/package/update-notifier#updatecheckinterval, also, it only shows once within that interval time. We might set it to ever 12 hours 🤔

**Comment by rauchg** [comment]
Can I see a screenshot of this? It'd be cool to use the same style as the friendly plugin

**Pull request closed without merging**